src/arith_uint256.h

```diff
@@ -174,7 +174,7 @@ class base_uint
     {
         // prefix operator
         int i = 0;
-        while (++pn[i] == 0 && i < WIDTH-1)
+        while (i < WIDTH && ++pn[i] == 0)
```

Comment:
> This template class can, although it probably won't, be instantiated with BITS < 32

Indeed - as the code throughout the entire class is implemented, it can only be instantiated for a positive multiple of 32. Might make sense to add a compile-time assertion for that.